homeassistant/components/fibaro/__init__.py

```diff
@@ -178,11 +182,31 @@ def _on_state_change(self, state: Any) -> None:
             for callback in self._callbacks[item]:
                 callback()
 
+        resolver = FibaroStateResolver(state)
+        for event in resolver.get_events():
+            fibaro_id = event.fibaro_id
+            if (
+                event.event_type.lower() == "centralsceneevent"
+                and fibaro_id in self._event_callbacks
+            ):
+                for callback in self._event_callbacks[fibaro_id]:
+                    callback(event)
+
     def register(self, device_id: int, callback: Any) -> None:
         """Register device with a callback for updates."""
         self._callbacks.setdefault(device_id, [])
         self._callbacks[device_id].append(callback)
 
+    def register_event(
```

Comment:
Is there no unregistration needed?

Reply:
As entities are always removed together with the integration I think it isn't needed to unregister it here.  The FibaroController which is the counterpart is correctly unregistered when the integration is unloaded. The rest of your comments are addressed in PR #101718.